### PR TITLE
Moved vendoring tool to dep.

### PR DESCRIPTION
**Pull request by dungeonmaster18**
Signed-off-by: dungeonmaster18 <umesh4257@gmail.com>

## Description

I have migrated the vendoring tool to dep from vndr.

## Motivation and Context
It is part of [openfaas/faas/issue#462](https://github.com/openfaas/faas/issues/462)

## How Has This Been Tested?

* So first i did ```dep init``` on the entire repo.
* Then to test my changes successfully worked or not I build a faas-cli binary using ```make build_redist```
* Output of binary test
```zsh
$ ./faas-cli
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|


Manage your OpenFaaS functions from the command line

Usage:
  faas-cli [flags]
  faas-cli [command]

Available Commands:
  build          Builds OpenFaaS function containers
  deploy         Deploy OpenFaaS functions
  help           Help about any command
  invoke         Invoke an OpenFaaS function
  list           List OpenFaaS functions
  login          Log in to OpenFaaS gateway
  logout         Log out from OpenFaaS gateway
  new            Create a new template in the current folder with the name given as name
  push           Push OpenFaaS functions to remote registry (Docker Hub)
  remove         Remove deployed OpenFaaS functions
  template       Downloads templates from the specified github repo
  version        Display the clients version information

Flags:
      --filter string   Wildcard to match with function names in YAML file
  -h, --help            help for faas-cli
      --regex string    Regex to match with function names in YAML file
  -f, --yaml string     Path to YAML file describing function(s)

Use "faas-cli [command] --help" for more information about a command.

```
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

/cc @alexellis @stefanprodan

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[[constraint]]
```
Comment (stefanprodan) [inline Gopkg.toml]: please add yaml, cobra, go-homedir and go-glob as constrains

Comment (dungeonmaster18) [reply Gopkg.toml]: okay I will do that

**Comment by stefanprodan** [comment]
"Then I changed the Gopkg.toml to remove unnecessary entries." why is that? every direct dependency must be in Gopkg.toml so we can easily upgrade it.

**Comment by dungeonmaster18** [comment]
@stefanprodan PTAL.

File: Gopkg.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  name = "github.com/mitchellh/go-homedir"
+  revision = "b8bc1bf767474819792c23f32d8286a45736f1c6"
+
+[[constraint]]
+  name = "github.com/morikuni/aec"
+  revision = "39771216ff4c63d11f5e604076f9c45e8be1067b"
+
+[[constraint]]
+  name = "github.com/openfaas/faas"
+  version = "0.5.6b-alpha"
```
Comment (stefanprodan) [inline Gopkg.toml]: this should be `0.6.16` I think https://github.com/openfaas/faas/tree/0.6.16

**Comment by dungeonmaster18** [comment]
It's autogenerated. I checked with the commit id and version they seem to
match.

**Comment by dungeonmaster18** [comment]
If I change that then I guess I have to add the vendor files with this as well.

**Comment by stefanprodan** [comment]
If it's autogenerated it's ok, probably the requests package didn't change in `0.6.16`

**Comment by dungeonmaster18** [comment]
Yup because it used the vndr.conf to generate. I was wondering should we update it to latest faas version

**Comment by stefanprodan** [comment]
Yes please try to update and rerun all tests

**Comment by dungeonmaster18** [comment]
Hey @stefanprodan PTAL, I updated the version of faas and tested it worked.

**Comment by dungeonmaster18** [comment]
Ping @alexellis , PTAL.

File: Gopkg.toml (hunk 1 above)
Comment (alexellis) [reply Gopkg.toml]: Has this been done now? @dungeonmaster18

**Comment by alexellis** [comment]
If we're all happy that the review comments are addressed I'll merge into master ready for the next release.

**Comment by dungeonmaster18** [comment]
All the review comments has been addressed @alexellis.

**Comment by alexellis** [comment]
@dungeonmaster18 due to #274 we need a quick rebase/update. Please could you take a look then I'll merge after a passing CI build.

**Comment by alexellis** [comment]
screen shot 2018-01-19 at 12 19 38

**Comment by dungeonmaster18** [comment]
Hey @alexellis I have fixed the merge conflict but someone has to remove the vendor.conf afterwards as it can't be removed here.

**Comment by alexellis** [comment]
You may be able to resolve by rebasing origin/master over the top. If you can't get it to rebase I can try to take a look at doing it locally with `git`. The UI won't let me rebase at all right now.

**Comment by dungeonmaster18** [comment]
Hey @alexellis rebased and everything seems to be working.

**Comment by alexellis** [comment]
LGTM thank you! Send me your photo for our contributors slide.

**Pull request merged**